frontend/src/hooks/useTaskRefresh.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+import {
+  TasksResponse,
+  fetchLeaderboard,
+  fetchTasks,
+  useTasks,
+} from "../api/backendComponents";
+import { LeaderboardEntry } from "../api/backendSchemas";
+import {
+  SignalRSocketEvent,
+  SignalRSocketHandler,
+} from "../utils/SignalRSocket";
+import config from "../config";
+
+export function useTasksWithRefresh() {
+  // We would benefit from converting this to a singleton
+  // to avoid multiple signalR subscriptions if hook is used multiple places.
+  const { data, isLoading, error } = useTasks({});
```

Comment:
On mobile so can't look at source: but Isn't useTasks a singleton with a "fetch tasks" method?

This change will force consumers to use useTaskRefresh instead of useTasks for reading tasks  which is not that intuitive

Reply:
Well, yes i guess. I just didn't want to touch the auto generated stuff as i am not at all familiar with it. So i just added sort of a "layer" inbetween

Reply:
Do you want me to move the signalR logic into the page itself? I don't really get what you want me to change here

Reply:
useTasks is auto generated i believe

Reply:
UseTasks is auto-generated yes, the way it is implemented today is good.

I would just expect that the auto-generated code included an exported "manually trigger fetch of all tasks method" which you could trigger when the signalR signal is received

Reply:
Do you mean to invalidate cache when signalr event is recieved?

Reply:
Yes there is a method fetchTasks, however i still don't get what is the problem? Consumers can still use the useTasks, its just for the page i wanted to move it out. I literally did it in the style the Leaderboard is written (almost). So do you want me to add the signalR stuff to the page itself?

Reply:
Yes invalidate cache.

So if you run fetchTasks, then the list of tasks in 'useTasks' will automatically update all consumers. Then there is no need for the useMemo.

Reply:
Ahhh didnt know that, so running fetchTasks will invalidate the cache automatically? Then i get it

Reply:
I'm not completely sure, but I would expect it to behave that way 🤞🏻

Reply:
Lol didn't work, very confusion